pkg/volume/cas_template_engine.go

```diff
@@ -86,56 +86,40 @@ type casCommon struct {
 	taskGroupRunner *task.TaskGroupRunner
 }
 
-// addTaskResultsToTaskResultTLP will add a run task's results to TaskResultTLP
-//
-// NOTE:
-//  This is a concrete implementation of task.PostTaskRunFn type.
-// Since task package does the low level execution, it has the results of
-// the execution as well the properties of resulting objects. This
-// function will be passed as a **closure** till task execution.
-
-// In other words it gets executed lazily post the run task's execution.
-//
-// NOTE:
-//  This will enable templating a run task template as follows:
-//
-// {{ .TaskResult.<Identity>.<prop1> }}
-// {{ .TaskResult.<Identity>.<prop2> }}
-//
-// NOTE:
-//  Above parsing scheme is translated by running `go template` against the run
-// task template
-func (c *casCommon) addTaskResultsToTaskResultTLP(taskResultsMap map[string]interface{}) {
-	if taskResultsMap == nil {
-		// nothing to do
-		return
-	}
-
-	// task results are mapped against the task's identity
-	for tID, tResults := range taskResultsMap {
-		// task result is added or updated to template values
-		// task result is set against TaskResultTLP.tID path of template values
-		util.SetNestedField(c.templateValues, tResults, string(v1alpha1.TaskResultTLP), tID)
+// buildCASCommon constructs and returns a casCommon structure based on the
+// given arguments
+func buildCASCommon(
+	casTemplate *v1alpha1.CASTemplate,
+	runtimeVals map[string]string,
+	fetcher task.TaskSpecFetcher,
+	grpRunner *task.TaskGroupRunner) casCommon {
+
+	return casCommon{
+		casTemplate: casTemplate,
+		templateValues: map[string]interface{}{
+			// runtime volume info is set against volume top level property
+			string(v1alpha1.VolumeTLP): runtimeVals,
```

Comment:
How can this be utilized in cstor pool contex?
Like for cstor pool we can have v1alpha1.CstorPool.TLP

Reply:
@sonasingh46 Can you try explaining the same via use of CASTemplate & RunTasks.
I am not too sure what is cstor & cstor pool etc.

Reply:
@AmitKumarDas 
Sure .